esp-wifi/src/tasks.rs

```diff
@@ -46,3 +46,10 @@ pub extern "C" fn timer_task(_param: *mut esp_wifi_sys::c_types::c_void) {
         yield_task();
     }
 }
+
+pub fn deinit_tasks() {
+    // Delete timer task
+    crate::preempt::delete_task(crate::preempt::current_task());
```

Comment:
how do we know the `current_task` is the timer task and the last remaining one is `main`?

Assuming the user calls deinit from main the current_task should be main and the remaining one is most probably the timer

Maybe just removing all remaining tasks is what we actually do and want